### PR TITLE
Updating repo onnx_cpu to adhere to vision team checklist

### PR DESCRIPTION
**Pull request by Emerald-Z**
Main contributions:

- setup linter
- updated Makefile appropriately (linting, testing)
- setup and tested Github Actions

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Build & publish module to registry 
+
+on:
+  push:
+    tags:
+    - '[0-9]+.[0-9]+.[0-9]+'
```
Comment (bhaney) [inline .github/workflows/publish.yml]: edit this to support -rc tags as well, e.g. `0.0.5-rc` used for testing and candidate purposes.

Comment (bhaney) [reply .github/workflows/publish.yml]: `[0-9]+.[0-9]+.[0-9]+(-rc)?` I think is the regex for that (the ? makes it optional)

**Review by bhaney — APPROVED** [review]
LGTM - the only thing is I would want you to check that this all works well by creating a new rc version tag and seeing if the deployed module works on darwin and arm64 linux (essentially finishing the last few checks on the check list)

File: .github/workflows/publish.yml (hunk 1 above)
Comment (Emerald-Z) [reply .github/workflows/publish.yml]: done!

**Pull request merged**